zmon_agent/discovery/kubernetes/cluster.py

```diff
@@ -334,14 +345,27 @@ def get_cluster_pods_and_containers(
         pod_entity.update(shared_properties)
         pod_entity.update(pod_labels)
         pod_entity.update(pod_annotations)
+        pod_entity.update(common_meta)
 
         for container in containers:
             container_name = container['name']
             container_image = container['image']
-            container_ready = container_statuses.get(container['name'], {}).get('ready', False)
-            container_restarts = container_statuses.get(container['name'], {}).get('restartCount', 0)
+
+            container_status = container_statuses.get(container_name, {})
+            container_ready = container_status.get('ready', False)
+            container_restarts = container_status.get('restartCount', 0)
             container_ports = [p['containerPort'] for p in container.get('ports', []) if 'containerPort' in p]
 
+            start_time = None
+            container_running = False
+
+            container_state = container_status.get("state", {})
+            if "running" in container_state:
+                start_time = container_state["running"]["startedAt"]
+                container_running = True
+            elif "terminated" in container_state:
+                start_time = container_state["running"]["startedAt"]
```

Comment:
should be `container_state["terminated"]["startedAt"]`

Reply:
Thanks! Fixed.